### PR TITLE
Handbook: update release docs

### PR DESCRIPTION
**Pull request by oandregal**
Updates release docs to:

- clarify schedule and dates section
- add a note about the required data one needs for the release

### CODE REVIEW

File: docs/contributors/release.md [hunk 1]
```diff
@@ -8,23 +8,19 @@ To release Gutenberg, you need commit access to the [WordPress.org plugin reposi
 
 ### Schedule
 
-We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date.
+We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date (note this is the day of _tagging the release candidate_).
```
Comment (ZebulanStanphill) [inline docs/contributors/release.md:11]:
```suggestion
We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date (the day when the release candidate is to be _tagged_).
```

File: docs/contributors/release.md [hunk 2]
```diff
@@ -8,23 +8,19 @@ To release Gutenberg, you need commit access to the [WordPress.org plugin reposi
...
+- **Two days after the first release candidate**, the stable version is created based on the last release candidate and any necessary regression fixes. Once the stable version is released, a post [like this](https://make.wordpress.org/core/2019/06/26/whats-new-in-gutenberg-26th-june/) describing the changes and performing a [performance audit](./testing-overview.md#performance-testing) is published.
 
 If critical bugs are discovered on stable versions of the plugin, patch versions can be released at any time.
 
 ### Release Tool
 
-The plugin release process is entirely automated. To release the RC version of the plugin, run the following command and follow the instructions: (Note that at the time of writing, the tool doesn't support releasing multiple consecutive RC releases)
+> Note that at the time of writing, the tool doesn't support releasing consecutive RC releases. However, it is possible to use the tool for patch releases following the first stable release.
```
Comment (ZebulanStanphill) [inline docs/contributors/release.md:21]: Using a blockquote here doesn't seem right to me.

Comment (nosolosw) [reply docs/contributors/release.md:21]: This is my thinking: we're using a similar approach in [other parts of the handbook](https://developer.wordpress.org/block-editor/developers/themes/block-based-themes/) to show notices (so they don't interrupt the main flow of the section but are visible). Does that make more sense? An alternative approach I'd be ok with is to add this note as a paragraph by the end of the section.

File: docs/contributors/release.md [hunk 3]
```diff
@@ -8,23 +8,19 @@ To release Gutenberg, you need commit access to the [WordPress.org plugin reposi
 
 ### Schedule
 
-We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date.
+We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date (the day when _the release candidate_ is to be _tagged).
```
Comment (draganescu) [inline docs/contributors/release.md:11]: > (the day when _the release candidate_ is to be _tagged).
(the day when _the release candidate_ is to be _tagged_).

Comment (nosolosw) [reply docs/contributors/release.md:11]:
```suggestion
We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date (the day when _the release candidate_ is to be tagged).
```

File: docs/contributors/release.md (hunk 2 above)
Comment (mkaz) [reply docs/contributors/release.md:21]: I think the blockquote is fine, it is used often for this case across our docs.
It is too bad we don't have gutenberg blocks in markdown ;-)

**Review by mkaz — APPROVED** [review]
Looks good 👍

File: docs/contributors/release.md (hunk 2 above)
Comment (ZebulanStanphill) [reply docs/contributors/release.md:21]: From an accessibility perspective, it's wrong to use blockquotes for things that aren't actually quotes. So if we want our docs to follow a11y guidelines, we should change all those instances of misused blockquotes at some point.

**Pull request merged**